homeassistant/components/workday/config_flow.py

```diff
@@ -202,10 +202,8 @@ async def async_step_options(
         if user_input is not None:
             combined_input: dict[str, Any] = {**self.data, **user_input}
 
-            if combined_input.get(CONF_COUNTRY, NONE_SENTINEL) == NONE_SENTINEL:
-                combined_input[CONF_COUNTRY] = None
-            if combined_input.get(CONF_PROVINCE, NONE_SENTINEL) == NONE_SENTINEL:
-                combined_input[CONF_PROVINCE] = None
+            combined_input.setdefault(CONF_COUNTRY, None)
```

Comment:
It's really not nice we have to do this kind of massaging everywhere, it would be much better if it was done by the schema.
Could we maybe add an allow_none parameter to `SelectSelectorConfig`?
```diff
-        vol.Optional(CONF_COUNTRY): SelectSelector(
+        vol.Optional(CONF_COUNTRYm default=None): SelectSelector(
             SelectSelectorConfig(
                 options=list(list_supported_countries()),
                 mode=SelectSelectorMode.DROPDOWN,
                 translation_key=CONF_COUNTRY,
+                allow_none=True,
             )
```

Reply:
Requires https://github.com/home-assistant/core/pull/101312 to be merged first as the PR add the new select option